mailet/standard/src/test/java/org/apache/james/transport/matchers/SenderHostIsTest.java

```diff
@@ -0,0 +1,219 @@
+        //Then
+        assertThat(senderHosts).contains("james.apache.org");
+        assertThat(senderHosts).contains("james2.apache.org");
+        assertThat(senderHosts).contains("james3.apache.org");
+        assertThat(senderHosts).contains("james4.apache.org");
+        assertThat(senderHosts).contains("james5.apache.org");
+    }
+
+    @Test
+    public void parseDomainsListShouldParseWhenCommaPattern() {
+        //Given
+        Collection<String> senderHosts;
+        //When
+        senderHosts = matcher.parseDomainsList("james.apache.org,james2.apache.org,james3.apache.org,james4.apache.org,james5.apache.org");
+        //Then
+        assertThat(senderHosts).contains("james.apache.org");
+        assertThat(senderHosts).contains("james2.apache.org");
+        assertThat(senderHosts).contains("james3.apache.org");
+        assertThat(senderHosts).contains("james4.apache.org");
+        assertThat(senderHosts).contains("james5.apache.org");
+    }
+
+    @Test
+    public void parseDomainsListShouldParseWhenSpacePattern() {
+        //Given
+        Collection<String> senderHosts;
+        //When
+        senderHosts = matcher.parseDomainsList("james.apache.org james2.apache.org james3.apache.org james4.apache.org james5.apache.org");
+        //Then
+        assertThat(senderHosts).contains("james.apache.org");
+        assertThat(senderHosts).contains("james2.apache.org");
+        assertThat(senderHosts).contains("james3.apache.org");
+        assertThat(senderHosts).contains("james4.apache.org");
+        assertThat(senderHosts).contains("james5.apache.org");
+    }
+
+    @Test
+    public void parseDomainsListShouldParseWhenMixedPatterns() {
+        //Given
+        Collection<String> senderHosts;
+        //When
+        senderHosts = matcher.parseDomainsList("james.apache.org james2.apache.org,james3.apache.org, james4.apache.org james5.apache.org");
+        //Then
+        assertThat(senderHosts).contains("james.apache.org");
+        assertThat(senderHosts).contains("james2.apache.org");
+        assertThat(senderHosts).contains("james3.apache.org");
+        assertThat(senderHosts).contains("james4.apache.org");
+        assertThat(senderHosts).contains("james5.apache.org");
+    }
+
+    @Test
+    public void parseDomainsListShouldIgnoreEmptyDomains() {
+        //Given
+        Collection<String> senderHosts;
+        //When
+        senderHosts = matcher.parseDomainsList("james.apache.org   james2.apache.org james3.apache.org , james4.apache.org,,,james5.apache.org");
+        //Then
+        assertThat(senderHosts.size()).isEqualTo(5);
+        assertThat(senderHosts).contains("james.apache.org");
+        assertThat(senderHosts).contains("james2.apache.org");
+        assertThat(senderHosts).contains("james3.apache.org");
+        assertThat(senderHosts).contains("james4.apache.org");
+        assertThat(senderHosts).contains("james5.apache.org");
```

Comment:
you can replace these 6 assertions by .containsOnly (and you can use the same in the other parseDomainsList test)